web3/eth/async_eth.py

```diff
@@ -0,0 +1,545 @@
+import asyncio
+from typing import (
+    Awaitable,
+    Callable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+)
+import warnings
+
+from eth_typing import (
+    Address,
+    BlockNumber,
+    ChecksumAddress,
+    HexStr,
+)
+from hexbytes import (
+    HexBytes,
+)
+
+from web3._utils.blocks import (
+    select_method_for_block_identifier,
+)
+from web3._utils.fee_utils import (
+    async_fee_history_priority_fee,
+)
+from web3._utils.filters import (
+    AsyncFilter,
+    select_filter_method,
+)
+from web3._utils.rpc_abi import (
+    RPC,
+)
+from web3.contract import (
+    AsyncContract,
+    AsyncContractCaller,
+)
+from web3.eth.base import (
+    BaseEth,
+)
+from web3.exceptions import (
+    OffchainLookup,
+    TimeExhausted,
+    TooManyRequests,
+    TransactionNotFound,
+)
+from web3.method import (
+    Method,
+    default_root_munger,
+)
+from web3.types import (
+    ENS,
+    BlockData,
+    BlockIdentifier,
+    BlockParams,
+    CallOverride,
+    FeeHistory,
+    FilterParams,
+    LogReceipt,
+    Nonce,
+    SyncStatus,
+    TxData,
+    TxParams,
+    TxReceipt,
+    Wei,
+    _Hash32,
+)
+from web3.utils import (
+    async_handle_offchain_lookup,
+)
+
+
+class AsyncEth(BaseEth):
+    is_async = True
+
+    _default_contract_factory: Type[
+        Union[AsyncContract, AsyncContractCaller]
+    ] = AsyncContract
+
+    # eth_accounts
+
+    _accounts: Method[Callable[[], Awaitable[Tuple[ChecksumAddress]]]] = Method(
```

Comment:
Since we're not using these underscore methods as shared variables between sync and async any more, what do you think about removing them and doing something like:

```py
def accounts(self) -> Tuple[ChecksumAddress]:
        return await Method(RPC.eth_accounts, is_property=True,)
```
It would save us some lines, but I don't have a strong preference either way

Reply:
We get rid of the type checking in this scenario, right?

Reply:
Right. You would know better since you're knee deep in it right now, but I don't think it would impact the typing since we still have the correct return typing in the method signature (i.e. `Tuple[ChecksumAddress]`)?

Reply:
I suppose for the properties this doesn't matter all that much since they don't take arguments but would we keep this pattern for the methods?

Reply:
Oops hadn't refreshed so didn't see your last comment. Yeah I think for properties this can make sense. I think keeping it for methods makes sense though.

Reply:
What do we gain from keeping it for the methods?

Reply:
Oh wait I see your point. I think I'm good with this change. I'll put it in a separate commit and we can go from there.

Reply:
@kclowes Ah, the way Method is set up is that the `__get__()` is what actually makes the call. So unless we can think of an elegant way to instantiate and get the `Method` all in the `return` call, I think the old way may still be the best way to go about things. Even when they lived in the same file, we still kept the underscored, typed definitions. Or am I missing something from your original comment?

Reply:
Ah, right. Nope, I had forgotten about the `__get__()` magic. Okay, this LGTM! Thanks for looking into it!